src/resources/uart_esp32.cc

```diff
@@ -262,7 +269,35 @@ class UartResourceGroup : public ResourceGroup {
   uint32 on_event(Resource* r, word data, uint32 state) override;
 };
 
-UART_ISR_INLINE void RxTxBuffer::return_buffer(uint8* buffer) {
+UART_ISR_INLINE uword RxTxBuffer::free_size_isr() {
```

Comment:
I am not sure this makes a difference as `portENTER_CRITICAL` and `portENTER_CRITICAL_ISR` on esp-idf are identical.